gemfirexd/tools/src/dunit/java/com/pivotal/gemfirexd/ddl/IndexPersistenceDUnit.java

```diff
@@ -618,4 +620,219 @@ public static void deleteAllOplogFiles() throws IOException {
       // ignore ...
     }
   }
+
+  private void verifyRowsIndexScanSNAP1933(Connection conn, String[] tableNames,
+      String[] indexNames, int maxValInsert, String alias) throws Exception {
+    final int delta1 = 100;
+    int sz = tableNames.length;
+    Statement s = conn.createStatement();
+    for (int i =0; i < sz; i++) {
+      ResultSet rs = s.executeQuery("select count(*) "
+          + alias
+          + " from ODS."
+          + tableNames[i]
+          + " --gemfirexd-properties index="
+          + indexNames[i]
+          + "");
+      while(rs.next()) {
+        Long l = rs.getLong(1);
+        assertTrue(i + ": Got value=" + l, l == maxValInsert);
+      }
+    }
+  }
+
+  private void verifyRowsTableScanSNAP1933(Connection conn, String[] tableNames,
+      int maxValInsert) throws Exception {
+    final int delta1 = 100;
+    int sz = tableNames.length;
+    Statement s = conn.createStatement();
+    for (int i =0; i < sz; i++) {
+      ResultSet rs = s.executeQuery("select count(*) "
+          + " from ODS."
+          + tableNames[i]
+          + " --gemfirexd-properties index=null"
+          + "");
+      while(rs.next()) {
+        Long l = rs.getLong(1);
+        assertTrue(i + ": Got value=" + l, l == maxValInsert);
+      }
+    }
+  }
+
+  private void updateRowsSNAP1933(Connection conn, String[] tableNames,
+      int delta) throws Exception {
+    final int delta1 = 100;
+    int sz = tableNames.length;
+    Statement s = conn.createStatement();
+    for (int i =0; i < sz; i++) {
+      for (int j = 0; j < 10; j++) {
+        int id = delta1 * j + 31 + delta;
+        int r = s.executeUpdate("update ODS."
+            + tableNames[i]
+            + " set cd_typ_ref_id=" + id + 10 + " where CD_VAL_ID=" +  id);
+        assert(r == 1);
+      }
+    }
+  }
+
+  private int restartServerSNAP1933(String[] tableNames, String[] indexNames, int maxValInsert,
+      int netPort, int restartServerNum, Properties serverInfo, int delta, String alias,
+      String persistIndexes) throws Exception {
+    Thread.sleep(3000);
+    stopVMNums(-restartServerNum);
+    Connection conn1 = TestUtil.getNetConnection(netPort, null, null);
+    updateRowsSNAP1933(conn1, tableNames, delta);
+    restartServerVMNums(new int[] { restartServerNum }, 0, null, serverInfo);
+    Thread.sleep(3000);
```

Comment:
Is a sleep required here or should it be rebalance_all_buckets to ensure redundancy is recovered?

Reply:
I think we should wait/sleep than re-balance buckets since customers may not be always be doing re-balance in case of HA. In current form test exposes more vulnerable situation.

Reply:
Ok, lets keep this way and discuss later. In normal use cases we do expect that if redundancy has been lost then users will start the new server with "-rebalance" option to restore it.